pyrefly/lib/lsp/non_wasm/server.rs

```diff
@@ -1097,7 +1114,8 @@ impl Server {
         params: crate::lsp::wasm::provide_type::ProvideTypeParams,
     ) -> Option<ProvideTypeResponse> {
         let uri = &params.text_document.uri;
-        let handle = self.make_handle_if_enabled(uri)?;
+        // provide_type is not a standard LSP service but we'll use Hover as it's similar
```

Comment:
I think it's confusing to disable this when hover is disabled. we should either never disable it or add a ProvideType LanguageService